### PR TITLE
Metricts collection with Stream API

### PR DESCRIPTION
**Pull request by desckapg**
**Core analytics and domain model:**

* Added the `MetricsCollector` class, which provides methods for key sales/customer analytics such as unique customer cities, total income from delivered orders, most popular product, average check for delivered orders, and identifying customers with more than five orders. (`MetricsCollector.java`)
* Defined domain model classes: `Order`, `Customer`, `OrderItem`, and enums `OrderStatus` and `Category` to represent orders, customers, items, and their statuses/categories. (`Order.java`, `Customer.java`, `OrderItem.java`, `OrderStatus.java`, `Category.java`)

**Testing and validation:**

* Added a comprehensive test suite for `MetricsCollector`, covering all analytics methods and edge cases such as empty input. (`MetricsCollectorTest.java`)

### CODE REVIEW

File: salescustomeranalysis/src/main/java/MetricsCollector.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                .mapToDouble(item -> item.getQuantity() * item.getPrice())
+                .sum();
+    }
+
+    /**
+     * Finds the most frequently ordered product by name across all orders.
+     *
+     * @return product name if present; empty if there are no items
+     */
+    public Optional<String> getMostPopularProduct() {
```
Comment (maxplyushko) [inline salescustomeranalysis/src/main/java/MetricsCollector.java]: method counts product occurrences but doesn't account for quantity. A product ordered once with quantity 100 should be more popular than a product ordered twice with quantity 1

File: salescustomeranalysis/src/main/java/MetricsCollector.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    private final List<Order> orders;
+
+    /**
+     * Creates a new metrics collector over the given orders.
+     *
+     * @param orders source orders to analyze; must be non-null
+     */
+    public MetricsCollector(List<Order> orders) {
+        this.orders = orders;
```
Comment (maxplyushko) [inline salescustomeranalysis/src/main/java/MetricsCollector.java]: there is no non-null validation, despite the Javadoc stating "must be non-null".

File: salescustomeranalysis/src/main/java/MetricsCollector.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    /**
+     * Returns customers who have placed more than five orders.
+     *
+     * @return list of customers with order count > 5
+     */
+    public List<Customer> getCustomersWhoHaveMoreThanFiveOrders() {
+        return orders.stream()
+                .map(Order::getCustomer)
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()))
```
Comment (maxplyushko) [inline salescustomeranalysis/src/main/java/MetricsCollector.java]: we can optimize grouping here to simplify the stream pipeline

File: salescustomeranalysis/src/main/java/MetricsCollector.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Collects analytical metrics over a list of orders.
+ * The instance holds a reference to the provided orders list.
+ */
+public class MetricsCollector {
```
Comment (maxplyushko) [inline salescustomeranalysis/src/main/java/MetricsCollector.java]: directory packages?? check style guide

**Review by maxplyushko — APPROVED** [review]
